### PR TITLE
Add buffer overload to Uri.produce and usage example

### LINKED ISSUE #354: Possible problem with URI URL encoder
The URL encoder in the `ocean.net.Uri` module seems to be encoding non-reserved characters in `IncScheme` mode which is not recommended in the specification (https://github.com/sociomantic-tsunami/ocean/blob/v3.x.x/src/ocean/net/Uri.d#L189).

For example `_` is not a reserved character but is encoded. I may have misunderstood what the `IncScheme` mode is supposed to do but to my mind it should follow the specification. See https://tools.ietf.org/html/rfc3986#section-2.2 for the reserved character list.

Note that the `EncodeNonUnreserved` class in `ocean.net.util.UrlEncoder` does the encoding correctly.

### PR DESCRIPTION
**Pull request by mihails-strasuns-sociomantic**
Fixes #354

### CODE REVIEW

File: src/ocean/net/Uri.d [hunk 1]
```diff
@@ -85,7 +86,7 @@ class Uri : UriView
         {"http",        80},
         {"http-ng",     80},
         {"https",       443},
-        l"imap",        143},
+        {"imap",        143},
```
Comment (nemanja-boric-sociomantic) [inline src/ocean/net/Uri.d]: Huh, what's this about?

Comment (nemanja-boric-sociomantic) [reply src/ocean/net/Uri.d]: Ah, typo introduced in the previous commit.

Comment (nemanja-boric-sociomantic) [reply src/ocean/net/Uri.d]: Ping + rebase.

**Comment by mihails-strasuns-sociomantic** [comment]
Rebased + moved typo fix into proper commit

**Comment by mihails-strasuns-sociomantic** [comment]
Ping @sociomantic-tsunami/core-team

File: src/ocean/net/Uri.d [hunk 2]
```diff
@@ -485,14 +496,9 @@ class Uri : UriView
 
     final override istring toString ()
     {
-        void[] s;
-        s.length = 256, s.length = 0;
-        enableStomping(s);
```
Comment (nemanja-boric-sociomantic) [inline src/ocean/net/Uri.d]: These three lines you've deleted are really interesting :+1:

**Review by nemanja-boric-sociomantic — APPROVED** [review]

**Comment by nemanja-boric-sociomantic** [comment]
:police_car: and it's good to go

**Comment by mihails-strasuns-sociomantic** [comment]
(notes added)

**Pull request merged**